item2/Dictionary.java

```diff
@@ -0,0 +1,53 @@
+import java.util.*;
+
+public class Dictionary {
+  private Map<String, Object> dict = new HashMap<String, Object>();
+
+  private Dictionary(Builder builder) {
+    dict = builder.dict;
```

Comment:
you are copying reference here. If the new elements are added to the builder in the future, this  reference changes to. Isn't Dictionary object supposed to be immutable.  If so you need to clone this here.

Reply:
Used deep cloning instead of pointing to reference here 9ff3963